File: ui/lanjii-ui/src/types/monitor/sysMenu.ts
/**
 * 系统菜单实体（对应后端SysMenu）
 */
export interface SysMenu {
  /** 菜单ID */
  id?: number
  /** 父菜单ID */
  parentId?: number
  /** 菜单名称 */
  name: string
  /** 菜单类型（1-目录，2-菜单，3-按钮） */
  type: number
  /** 路由路径(前端路由或外链URL) */
  path?: string
  /** 组件路径 */
  component?: string
  /** 权限标识 */
  permission?: string
  /** 菜单图标 */
  icon?: string
  /** 显示顺序 */
  sortOrder?: number
  /** 是否可见（0-隐藏，1-显示） */
  isVisible?: number
  /** 是否启用（1启用 0禁用） */
  isEnabled?: number
  /** 是否外链（0-否，1-是） */
  isExt?: number
  /** 打开方式（0-内嵌，1-新窗口） */
  openMode?: number
  /** 子菜单列表 */
  children?: SysMenu[]
  /** 启用状态标签（字典转换） */
  isEnabledLabel?: string
  /** 可见状态标签（字典转换） */
  isVisibleLabel?: string
  /** 创建时间 */
  createTime?: string
  /** 更新时间 */
  updateTime?: string
  /** 创建人 */
  createBy?: string
  /** 更新人 */
  updateBy?: string
  /** 删除标志（0-未删除，1-已删除） */
  deleted?: number
}

File: src/main/java/com/lanjii/biz/admin/system/model/entity/SysMenu.java
package com.lanjii.biz.admin.system.model.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lanjii.biz.admin.system.model.dto.SysMenuDTO;
import com.lanjii.biz.admin.system.model.vo.SysMenuVO;
import com.lanjii.core.base.BaseEntity;
import com.lanjii.core.base.BaseEntityMapper;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

/**
 * 菜单表(SysMenu)表实体类
 *
 * @author lanjii
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_menu")
public class SysMenu extends BaseEntity<SysMenu> {

    /**
     * 菜单ID
     */
    private Long id;

    /**
     * 父菜单ID
     */
    private Long parentId;

    /**
     * 祖级列表(逗号分隔)
     */
    private String ancestors;

    /**
     * 菜单名称
     */
    private String name;

    /**
     * 菜单类型（1-目录，2-菜单，3-按钮）
     */
    private Integer type;

    /**
     * 路由路径(前端路由或外链URL)
     */
    private String path;

    /**
     * 组件路径
     */
    private String component;

    /**
     * 权限标识
     */
    private String permission;

    /**
     * 菜单图标
     */
    private String icon;

    /**
     * 显示顺序
     */
    private Integer sortOrder;

    /**
     * 是否可见（0-隐藏，1-显示）
     */
    private Integer isVisible;

    /**
     * 是否启用（1启用 0禁用）
     */
    private Integer isEnabled;

    /**
     * 是否外链（0-否，1-是）
     */
    private Integer isExt;

    /**
     * 打开方式（0-内嵌，1-新窗口）
     */
    private Integer openMode;

    @Mapper
    public interface SysMenuMapper extends BaseEntityMapper<SysMenu, SysMenuVO, SysMenuDTO> {

        @Mapping(target = "children", ignore = true)
        @Mapping(target = "isEnabledLabel", expression = "java(dictValueToLabel(entity.getIsEnabled(),\"STATUS\"))")
        @Mapping(target = "isVisibleLabel", expression = "java(dictValueToLabel(entity.getIsVisible(),\"VISIBLE\"))")
        SysMenuVO toVo(SysMenu entity);

        @Mapping(target = "ancestors", ignore = true)
        SysMenu toEntity(SysMenuDTO dto);

    }

    public static final SysMenuMapper INSTANCE = Mappers.getMapper(SysMenuMapper.class);

}